src/mg_decoration.cpp

```diff
@@ -338,16 +338,63 @@ DecoSchematic::DecoSchematic()
 }
 
 
-size_t DecoSchematic::generate(MMVManip *vm, PcgRandom *pr, v3s16 p)
+bool DecoSchematic::canPlaceSchematicDecoration(MMVManip *vm, v3s16 p)
 {
 	// Schematic could have been unloaded but not the decoration
 	// In this case generate() does nothing (but doesn't *fail*)
 	if (schematic == NULL)
-		return 0;
+		return false;
 
+	// Check if the decoration can be placed on this node
 	u32 vi = vm->m_area.index(p);
-	content_t c = vm->m_data[vi].getContent();
-	if (!CONTAINS(c_place_on, c))
+	if (!CONTAINS(c_place_on, vm->m_data[vi].getContent()))
+		return false;
+
+	// Don't continue if there are no spawnby constraints
+	if (nspawnby == -1)
+		return true;
+
+	int nneighs = 0;
+	static const v3s16 dirs[16] = {
+		v3s16( 0, 0,  1),
+		v3s16( 0, 0, -1),
+		v3s16( 1, 0,  0),
+		v3s16(-1, 0,  0),
+		v3s16( 1, 0,  1),
+		v3s16(-1, 0,  1),
+		v3s16(-1, 0, -1),
+		v3s16( 1, 0, -1),
+
+		v3s16( 0, 1,  1),
```

Comment:
I just noticed some subtle API breakage:
When you added this second plane to check, you inadvertently made older mods twice as likely to spawn a decoration (since there are twice as many nodes being checked, but num_spawn_by stays the same).
I do not have any suggestions on how to solve this.